src/python/pants/backend/python/subsystems/twine.py

```diff
@@ -95,18 +82,33 @@ def config_request(self) -> ConfigFilesRequest:
             check_existence=[".pypirc"],
         )
 
-    def ca_certs_digest_request(self, default_ca_certs_path: str | None) -> CreateDigest | None:
-        ca_certs_path: str | None = self.ca_certs_path
-        if ca_certs_path == "<inherit>":
-            ca_certs_path = default_ca_certs_path
-        if not ca_certs_path or ca_certs_path == "<none>":
-            return None
-
-        # The certs file will typically not be in the repo, so we can't digest it via a PathGlobs.
-        # Instead we manually create a FileContent for it.
-        ca_certs_content = Path(ca_certs_path).read_bytes()
-        chrooted_ca_certs_path = os.path.basename(ca_certs_path)
-        return CreateDigest((FileContent(chrooted_ca_certs_path, ca_certs_content),))
+    class EnvironmentAware:
+        ca_certs_path = StrOption(
+            advanced=True,
+            default="<inherit>",
+            help=softwrap(
+                """
+                Path to a file containing PEM-format CA certificates used for verifying secure
+                connections when publishing python distributions.
+
+                Uses the value from `[GLOBAL].ca_certs_path` by default. Set to `"<none>"` to
+                not use the default CA certificate.
+                """
+            ),
+        )
+
+        def ca_certs_digest_request(self, default_ca_certs_path: str | None) -> CreateDigest | None:
+            ca_certs_path: str | None = self.ca_certs_path
+            if ca_certs_path == "<inherit>":
+                ca_certs_path = default_ca_certs_path
+            if not ca_certs_path or ca_certs_path == "<none>":
+                return None
+
+            # The certs file will typically not be in the repo, so we can't digest it via a PathGlobs.
+            # Instead we manually create a FileContent for it.
+            ca_certs_content = Path(ca_certs_path).read_bytes()
```

Comment:
This is not safe, but can be fixed in a follow up https://github.com/pantsbuild/pants/issues/16987

Reply:
It's also fundamentally not correct :)

Happy to leave it as